src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -98,6 +98,13 @@ static bool IsCompatible(Type t, Type source)
 
         static bool IsTypeATimeoutHandledByAnySaga(Type type, IEnumerable<Type> sagas)
         {
+            // MakeGenericType() throws an exception if passed a ref struct type
+            // Messages cannot be ref struct types
+            if (type.CustomAttributes.Any(a => a.AttributeType.Name == "IsByRefLikeAttribute"))
```

Comment:
```suggestion
            if (type.CustomAttributes.Any(a => a.AttributeType.FullName == "System.Runtime.CompilerServices.IsByRefLikeAttribute"))
```

Would be better I think because even when the type is "shimed" it has to be in that namespace as far as I recall. 

I think it is fine because we preselect the types anyway to be saga types and only then we go through the handler types right?

Is this ever executed in a reflection only context? I don't think so right?

Reply:
The question about reflection only context comes from my blurry knowledge that CustomAttributes instantiates the attributes which means the constructors are called. This might not always work or might be more expensive compared to getting the data only which could be achieved with 

```suggestion
            if (type.GetCustomAttributesData().Any(attr => attr.AttributeType.FullName == "System.Runtime.CompilerServices.IsByRefLikeAttribute"))
```